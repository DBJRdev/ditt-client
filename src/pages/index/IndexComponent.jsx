import ImmutablePropTypes from 'react-immutable-proptypes';
import PropTypes from 'prop-types';
import React from 'react';
import WorkLogCalendar from '../../components/WorkLogCalendar';

class IndexComponent extends React.Component {
  componentDidMount() {
    this.props.fetchWorkLogList();
  }

  render() {
    return (
      <div>
        <h2>Work log</h2>
        {this.props.isFetching ? 'Loading...' : <WorkLogCalendar workLogList={this.props.workLogList} />}
      </div>
    );
  }
}

IndexComponent.propTypes = {
  fetchWorkLogList: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
  workLogList: ImmutablePropTypes.listOf(ImmutablePropTypes.mapContains({
    endTime: PropTypes.shape.isRequired,
    id: PropTypes.number.isRequired,
    startTime: PropTypes.shape.isRequired,
  })).isRequired,
};

export default IndexComponent;
