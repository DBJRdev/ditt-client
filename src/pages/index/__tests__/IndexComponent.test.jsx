import Immutable from 'immutable';
import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import IndexComponent from '../IndexComponent';

let fakeTimestamp = null;

beforeEach(() => {
  fakeTimestamp = sinon.useFakeTimers(1514764800000);
});

afterEach(() => {
  fakeTimestamp.restore();
});

describe('rendering', () => {
  it('renders correctly', () => {
    const props = {
      addWorkLog: () => {},
      deleteWorkLog: () => {},
      fetchWorkHoursList: () => {},
      fetchWorkMonth: () => {},
      fetchWorkMonthList: () => new Promise(() => {}),
      history: {
        push: () => {},
      },
      isFetching: false,
      isPosting: false,
      logout: () => {},
      markWaitingForApproval: () => {},
      uid: 1,
      workHoursList: Immutable.List(),
      workMonth: null,
      workMonthList: Immutable.List(),
    };
    const tree = shallow(<IndexComponent {...props} />);

    expect(tree).toMatchSnapshot();
  });
});
