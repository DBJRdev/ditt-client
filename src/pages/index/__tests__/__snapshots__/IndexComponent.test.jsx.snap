// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<Connect(LayoutComponent)
  loading={false}
  title="Work logs"
>
  <WorkLogCalendar
    addWorkLog={[Function]}
    changeSelectedDate={[Function]}
    deleteWorkLog={[Function]}
    isPosting={false}
    markWaitingForApproval={[Function]}
    selectedDate={"2018-01-01T00:00:00.000Z"}
    workHoursList={Immutable.List []}
    workMonth={null}
    workMonthList={Immutable.List []}
  />
</Connect(LayoutComponent)>
`;
